Merge feature extraction and main environment

clane9 opened the pull request.

clane9 committed:
merge feature extraction environment with main env

- upgraded torch to 2.7.1 to fix an incompatibility with flash attention
- also added citation to readme

clane9 committed:
actually, can install flash-attn the usual way now

since we upgraded torch, can just install flash-attn the usual way from
pypi (I think).

clane9 commented:
@mihirneal I think I resolved the environment issues. Just upgraded torch to 2.7.1. Also some updates to the readme. Feel free to edit/merge. And then we'll merge your #4 into main.

mihirneal committed:
merged feature extraction class

mihirneal commented:
thanks! Flash attention works now. I've also merged the FeatureExtraction class across models.

clane9 committed:
exclude feature extraction scripts from ruff

clane9 committed:
update `FeatureExtractor`

clane9 committed:
update feature extraction scripts

- minor changes to feature extractor api
- fix trailing whitespace (but don't run ruff)
